aicsimageio/aics_image.py

```diff
@@ -115,67 +121,76 @@ class AICSImage:
     """
 
     # Construct SUPPORTED_READERS as we parse FORMAT_IMPLEMENTATIONS
-    SUPPORTED_READERS: List[ReaderType] = []
-    for reader_path in FORMAT_IMPLEMENTATIONS.values():
-        module_path, reader_name = reader_path.rsplit(".", 1)
-        try:
-            loaded_mod = importlib.import_module(module_path)
-            ReaderClass = getattr(loaded_mod, reader_name)
-            if ReaderClass not in SUPPORTED_READERS:
-                SUPPORTED_READERS.append(ReaderClass)
-
-        except ImportError:
-            pass
+    SUPPORTED_READERS: List[Type[Reader]] = []
+    for reader_paths in FORMAT_IMPLEMENTATIONS.values():
+        for reader_path in reader_paths:
+            try:
+                ReaderClass = _load_reader(reader_path)
+                if ReaderClass not in SUPPORTED_READERS:
+                    SUPPORTED_READERS.append(ReaderClass)
+
+            except ImportError:
+                pass
 
     @staticmethod
-    def determine_reader(image: types.ImageLike, **kwargs: Any) -> ReaderType:
+    def determine_reader(image: types.ImageLike, **kwargs: Any) -> Type[Reader]:
         """
         Cheaply check to see if a given file is a recognized type and return the
         appropriate reader for the image.
 
         Returns
         -------
-        ReaderClass: ReaderType
+        ReaderClass: Type[Reader]
             The reader that supports the provided image.
 
         Raises
         ------
         exceptions.UnsupportedFileFormatError
             No reader could be found that supports the provided image.
         """
-        for ReaderClass in AICSImage.SUPPORTED_READERS:
-            if ReaderClass.is_supported_image(image, **kwargs):  # type: ignore
-                return ReaderClass
-
-        # Construct non-URI image "paths"
-        # numpy, dask, etc.
+        # Try reader detection based off of file path extension
         if isinstance(image, (str, Path)):
             path = str(image)
 
             # Check for extension in FORMAT_IMPLEMENTATIONS
-            # We do a reverse string match, i.e. match from the back
-            # of the path.
-            # It is unfortunately pretty common for microscopy images to be labeled
-            # "some-file.czi.ome.tiff" where the file _was_ a CZI
-            # but was converted to OME-TIFF.
-            for format_ext in FORMAT_IMPLEMENTATIONS.keys():
-                if path[(len(format_ext) * -1) :] == format_ext:
+            for format_ext, readers in FORMAT_IMPLEMENTATIONS.items():
```

Comment:
this feels wrong to have to loop over every single entry in FORMAT_IMPLEMENTATIONS

Reply:
We've talked about this before. It's because we can't just split by "." because we have file formats like ".ome.tiff" and "something.something" so we cant find the suffix that way. Then because of that, we can't just do a lookup. :shrug:

Reply:
I wonder if ome.tiff is the only example of that.  Are there any others?
I would happily special-case tiff+ome.tiff if it made everything else cleaner and more elegant.  Anyway, I know what is here now is working so I approved it :)